aggregator_core/src/datastore.rs

```diff
@@ -3608,17 +3608,14 @@ impl<C: Clock> Transaction<'_, C> {
                 "WITH batches AS (
                     SELECT
                         outstanding_batches.batch_id AS batch_id,
-                        COUNT(DISTINCT report_aggregations.client_report_id) AS count
+                        SUM(batch_aggregations.report_count) AS count
                     FROM outstanding_batches
-                    JOIN aggregation_jobs
-                      ON aggregation_jobs.task_id = (SELECT id FROM tasks WHERE task_id = $1)
-                     AND aggregation_jobs.batch_id = outstanding_batches.batch_id
-                    JOIN report_aggregations
-                      ON report_aggregations.aggregation_job_id = aggregation_jobs.id
-                     AND report_aggregations.state = 'FINISHED'
-                    GROUP BY outstanding_batches.batch_id
+                    JOIN batch_aggregations
+                      ON batch_aggregations.batch_identifier = outstanding_batches.batch_id
+                    WHERE outstanding_batches.task_id = (SELECT id FROM tasks WHERE task_id = $1)
+                    GROUP BY outstanding_batches.batch_id, batch_aggregations.aggregation_param
```

Comment:
I'm not sure grouping by aggregation param is helpful; `outstanding_batches` is scoped to the batch (i.e. `(batch_id)` only) rather than batch & aggregation param (i.e. `(batch_id, aggregation_param)`). We'll need to fix this up once we support aggregating the same batch multiple times, but for now I don't think we should look to the aggregation param at all.

Reply:
Both the aggregation_param GROUP BY and the SELECT DISTINCT were intended to be parallel to the `COUNT(DISTINCT ...)` in the old query, for this purpose. I can take them out.